src/bin.ts

```diff
@@ -134,6 +134,29 @@ export const deploy = async ({ yes, bucket, userAgent }: DeployArguments = {}) =
         if (process.env.HTTP_PROXY) {
             httpOptions = {
                 agent: proxy(process.env.HTTP_PROXY),
+                ...httpOptions,
+            };
+        }
+
+        if (config.timeout !== undefined) {
+            httpOptions = {
+                timeout: config.timeout,
```

Comment:
This section seems a bit verbose. What happens if we provide `undefined` to S3 instead of excluding a property completely?

Can we just do:
```javascript
httpOptions = {
    timeout: config.timeout,
    connectTimeout: config.connectTimeout,
    customBackoff: config.fixedRetryDelay !== undefined ? () => config.fixedRetryDelay : undefined,
    ...httpOptions
```
or would this not work?

Reply:
It would, I was just trying to follow suit on how the proxy agent was set. I changed it to a more compact format but I kept the proxy agent separate because TS was giving me issues with it not having all required props - is that why it was added like this before?